tsurud root-user-create: nil-pointer panic under non-UserScheme auth (oidc) and duplicate-key error after successful update

Two bugs in the same function (`createRootUserCmd.Run`, `cmd/tsurud/token.go`),
both reproducible on 1.28.x through current master (the file is unchanged
between them).

**Bug A — panic when the user doesn't exist and `auth:scheme` is a scheme
that doesn't implement `auth.UserScheme` (e.g. `oidc`):**

```
$ tsurud root-user-create new@example.com
panic: runtime error: invalid memory address or nil pointer dereference
github.com/tsuru/tsuru/auth.(*User).Create(0x0, ...)  auth/user.go
main.createRootUserCmd.Run({}, ...)                   cmd/tsurud/token.go
```

`auth.GetUserByEmail` returns `(nil, err)` for a nonexistent user; only the
`native` and `oauth` schemes implement `auth.UserScheme`, so for `oidc` the
code reaches the `else` branch and calls `user.Create(ctx)` on the nil `user`.

**Bug B — missing `return` after the existing-user path:**

When the user *does* exist, the super role is added, "Root user successfully
updated." is printed… and then execution falls through to the creation code
anyway:

```
$ tsurud root-user-create existing@example.com
Root user successfully updated.
Error: write exception: ... E11000 duplicate key error collection: tsuru.users
index: email_1 dup key: { email: "existing@example.com" }
```

Exit code 1 even though the operation succeeded (breaks scripted/idempotent
bootstrap flows).

Repro: any tsurud with `auth: {scheme: oidc}` — run `root-user-create` with a
new email (panics); run it twice with the same email (E11000). PR incoming.

--- a/cmd/tsurud/testdata/tsuru-oidc.conf
+++ b/cmd/tsurud/testdata/tsuru-oidc.conf
@@ -1,0 +1,18 @@
+listen: "0.0.0.0:8080"
+host: http://tsuru.server:8080/
+use-tls: false
+tls:
+  cert-file: /path/to/cert.pem
+  key-file: /path/to/key.pem
+database:
+  url: 127.0.0.1:27017
+  name: tsuru_tsurud_tests
+auth:
+  scheme: oidc
+  salt: tsuru-salt
+  token-expire-days: 2
+  hash-cost: 4
+  user-registration: true
+quota:
+  units-per-app: 4
+  apps-per-user: 2

--- a/cmd/tsurud/token.go
+++ b/cmd/tsurud/token.go
@@ -41,6 +41,7 @@ func (createRootUserCmd) Run(c *cmd.Context) error {
 			return err
 		}
 		fmt.Fprintln(c.Stdout, "Root user successfully updated.")
+		return nil
 	}
 	var confirm, password string
 	if scheme == nativeSchemeName {

--- a/cmd/tsurud/token.go
+++ b/cmd/tsurud/token.go
@@ -69,6 +69,7 @@ func (createRootUserCmd) Run(c *cmd.Context) error {
 			return err
 		}
 	} else {
+		user = &auth.User{Email: email}
 		err = user.Create(ctx)
 		if err != nil {
 			return err

--- a/cmd/tsurud/token_test.go
+++ b/cmd/tsurud/token_test.go
@@ -41,3 +41,25 @@ func (s *S) TestCreateRootUserCmdRun(c *check.C) {
 	c.Assert(perms[0].Scheme, check.Equals, permission.PermUser)
 	c.Assert(perms[1].Scheme, check.Equals, permission.PermAll)
 }
+
+func (s *S) TestCreateRootUserCmdRunNonUserSchemeNewUser(c *check.C) {
+	var stdout, stderr bytes.Buffer
+	context := cmd.Context{
+		Args:   []string{"my@user.com"},
+		Stdout: &stdout,
+		Stderr: &stderr,
+		Stdin:  strings.NewReader(""),
+	}
+	command := &tsurudCommand{Command: createRootUserCmd{}}
+	command.Flags().Parse(true, []string{"--config", "testdata/tsuru-oidc.conf"})
+	err := command.Run(&context)
+	c.Assert(err, check.IsNil)
+	c.Assert(stdout.String(), check.Equals, "Root user successfully created.\n")
+	u, err := auth.GetUserByEmail(stdContext.TODO(), "my@user.com")
+	c.Assert(err, check.IsNil)
+	perms, err := u.Permissions(stdContext.TODO())
+	c.Assert(err, check.IsNil)
+	c.Assert(perms, check.HasLen, 2)
+	c.Assert(perms[0].Scheme, check.Equals, permission.PermUser)
+	c.Assert(perms[1].Scheme, check.Equals, permission.PermAll)
+}

--- a/cmd/tsurud/token_test.go
+++ b/cmd/tsurud/token_test.go
@@ -42,6 +42,31 @@ func (s *S) TestCreateRootUserCmdRun(c *check.C) {
 	c.Assert(perms[1].Scheme, check.Equals, permission.PermAll)
 }
 
+func (s *S) TestCreateRootUserCmdRunExistingUser(c *check.C) {
+	existing := &auth.User{Email: "my@user.com"}
+	err := existing.Create(stdContext.TODO())
+	c.Assert(err, check.IsNil)
+	var stdout, stderr bytes.Buffer
+	context := cmd.Context{
+		Args:   []string{"my@user.com"},
+		Stdout: &stdout,
+		Stderr: &stderr,
+		Stdin:  strings.NewReader("foo123\nfoo123\n"),
+	}
+	command := &tsurudCommand{Command: createRootUserCmd{}}
+	command.Flags().Parse(true, []string{"--config", "testdata/tsuru.conf"})
+	err = command.Run(&context)
+	c.Assert(err, check.IsNil)
+	c.Assert(stdout.String(), check.Equals, "Root user successfully updated.\n")
+	u, err := auth.GetUserByEmail(stdContext.TODO(), "my@user.com")
+	c.Assert(err, check.IsNil)
+	perms, err := u.Permissions(stdContext.TODO())
+	c.Assert(err, check.IsNil)
+	c.Assert(perms, check.HasLen, 2)
+	c.Assert(perms[0].Scheme, check.Equals, permission.PermUser)
+	c.Assert(perms[1].Scheme, check.Equals, permission.PermAll)
+}
+
 func (s *S) TestCreateRootUserCmdRunNonUserSchemeNewUser(c *check.C) {
 	var stdout, stderr bytes.Buffer
 	context := cmd.Context{

